src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs

```diff
@@ -63,7 +63,29 @@ public async Task<DbConnectionValidatorResult> ValidateAsync(string databaseProv
             return DbConnectionValidatorResult.InvalidConnection;
         }
 
-        var factory = GetFactory(providerName, connectionString);
+        var (factory, dialect) = GetConnectionFactoryAndSqlDialect(providerName, connectionString);
+
+        switch (providerName)
```

Comment:
This can be removed now since you created `GetConnectionFactoryAndSqlDialect`

Reply:
Seems I'm out of coverage at that time ;)